src/EtherFiAdmin.sol

```diff
@@ -164,6 +170,31 @@ contract EtherFiAdmin is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         emit AdminOperationsExecuted(msg.sender, reportHash);
     }
 
+    // Process `liquidityPool.batchApproveRegistration` for the last task in the pending queue
+    function AdminTask_ApproveValidators(bytes32 _reportHash, uint256[] memory _validatorsToApprove, bytes[] calldata _pubKey, bytes[] calldata _signature) external isAdmin() {
+        AdminTask memory task = _dequeuePendingTask("validatorsToApprove");
+
+        require(etherFiOracle.isConsensusReached(_reportHash), "EtherFiAdmin: report didn't reach consensus");
+        require(task.operationHash == keccak256(abi.encode(_reportHash, _validatorsToApprove, _pubKey, _signature)), "EtherFiAdmin: invalid task");
+
+        liquidityPool.batchApproveRegistration(_validatorsToApprove, _pubKey, _signature);
+    }
+
+    // FIFO is better?
+    // should we support random-access so that the admin can process the pending transactions in any order?
+    function _dequeuePendingTask(string memory _operation) internal returns (AdminTask memory) {
```

Comment:
maybe this should be an external function with onlyAdmin modifier. What happens if admin wants to unpublished? How do we clear the queue?

Reply:
good point. we need to add a function to handle such cases